src/WorkBC.Data/Migrations/20240430195843_UpdateNocCodes2016Data.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Text;
+using System;
+using WorkBC.Data.Model.Enterprise;
+
+#nullable disable
+
+namespace WorkBC.Data.Migrations
+{
+    public partial class UpdateNocCodes2016Data : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            //Add new columns to dbo.NocCodes2021 tables.
+            migrationBuilder.AddColumn<string>(
+            name: "Code2016",
+            table: "NocCodes2021",
+            type: "nvarchar(30)",
+            maxLength: 30,
+            nullable: true);
+
+            var allNocs = GetNocCodes2016();
+
+            foreach (var noc in allNocs)
+            {
+                if (!String.IsNullOrEmpty(noc.noc_2021))
+                {                    
+                    string code2016 = noc.noc_2016.ToString();
+
+                    //The field 'Id' is derived from the field 'Code' and is not an auto-generated Identity field.
+                    int.TryParse(noc.noc_2021.ToString(), out int id);
+
+                    //Updating Code2016 column in the table dbo.NocCodes2021.
+                    migrationBuilder.UpdateData(
+                    table: "NocCodes2021",
+                    keyColumn: "Id",
+                    keyValue: id,
+                    column: "Code2016",
+                    value: code2016);
+                }
+            }
+
+        }
+
+        protected override void Down(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropColumn(
+            name: "Code2016",
+            table: "NocCodes2021");
+        }
+
+        public List<NocCodeSSOT1> GetNocCodes2016()
+        {
+
+            List<NocCodeSSOT1> newSSOTCodes = new List<NocCodeSSOT1>();
+            //Read the ssot_nocs json file from the path:~\workbc-jb\src\WorkBC.Web
+            string jsonString = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, "ssot_nocs.json"));
+            // convert string to stream
+            byte[] byteArray = Encoding.UTF8.GetBytes(jsonString);
+            MemoryStream jsonStream = new MemoryStream(byteArray);
+
+            newSSOTCodes = JsonSerializer.Deserialize<List<NocCodeSSOT1>>(jsonStream);
+
+            return newSSOTCodes;
+
+        }
+    }
+
+    public class NocCodeSSOT1
```

Comment:
Can we make this class (and the one in the other migration) private, and therefore not need to call them differently?

Reply:
Cannot make this private. Elements defined in a namespace cannot be explicitly declared as private, protected, protected internal or private protected (Compiler Error CS1527). These 2 classes can only be made public or internal. In either scenario, they are within the same namespace and so will have to be named differently. I have renamed the "NocCodeSSOT1" to "NocCodeSSOTwith2016" for aesthetic purpose. :)

Reply:
Haha thanks for taking care of the aesthetics of the codebase :smile: